cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/UsernamePasswordWrapperAuthenticationHandler.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.support.pac4j.authentication.handler.support;
+
+import org.jasig.cas.authentication.Credential;
+import org.jasig.cas.authentication.PreventedException;
+import org.jasig.cas.authentication.UsernamePasswordCredential;
+import org.jasig.cas.authentication.handler.NoOpPrincipalNameTransformer;
+import org.jasig.cas.authentication.handler.PasswordEncoder;
+import org.jasig.cas.authentication.handler.PlainTextPasswordEncoder;
+import org.jasig.cas.authentication.handler.PrincipalNameTransformer;
+import org.pac4j.core.credentials.Credentials;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+
+import javax.security.auth.login.AccountNotFoundException;
+import javax.validation.constraints.NotNull;
+import java.security.GeneralSecurityException;
+
+/**
+ * Pac4j authentication handler which works on a CAS username / password credential
+ * and uses a pac4j authenticator and profile creator to play authentication.
+ *
+ * @author Jerome Leleu
+ * @since 4.2.0
+ */
+public class UsernamePasswordWrapperAuthenticationHandler extends AbstractWrapperAuthenticationHandler {
```

Comment:
Sounds good. 

- Misagh

> On Oct 16, 2015, at 1:14 AM, LELEU Jérôme <notifications@github.com> wrote:
> 
> In cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/authentication/handler/support/UsernamePasswordWrapperAuthenticationHandler.java <https://github.com/Jasig/cas/pull/1217#discussion_r42214579>:
> 
> > +import org.jasig.cas.authentication.handler.PrincipalNameTransformer;
> > +import org.pac4j.core.credentials.Credentials;
> > +import org.pac4j.http.credentials.UsernamePasswordCredentials;
> > +
> > +import javax.security.auth.login.AccountNotFoundException;
> > +import javax.validation.constraints.NotNull;
> > +import java.security.GeneralSecurityException;
> > +
> > +/**
> > + * Pac4j authentication handler which works on a CAS username / password credential
> > + * and uses a pac4j authenticator and profile creator to play authentication.
> > + *
> > + * @author Jerome Leleu
> > + * @since 4.2.0
> > + */
> > +public class UsernamePasswordWrapperAuthenticationHandler extends AbstractWrapperAuthenticationHandler {
> The AbstractWrapperAuthenticationHandler is generic, depending on CAS and pac4j credentials.
> 
> This UsernamePasswordWrapperAuthenticationHandler is the most useful handler where CAS username/pwd credentials are converted into pac4j username/pwd credentials.
> The only left use case is for pac4j string token credentials (one string as the credentials), but I didn't know from which CAS credentials we could convert this.
> 
> I'm open to any suggestion?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Jasig/cas/pull/1217/files#r42214579>.
>